### PR TITLE
Include invalid name/value when raising TypeError in DataField

### PR DESCRIPTION
**Pull request by plotski**
I couldn't find a straight-forward way to test invalid values.

### CODE REVIEW

File: httpx/_multipart.py [hunk 1]
```diff
@@ -19,9 +19,11 @@ class DataField:
 
     def __init__(self, name: str, value: typing.Union[str, bytes]) -> None:
         if not isinstance(name, str):
-            raise TypeError("Invalid type for name. Expected str.")
+            raise TypeError(f"Invalid type for name. Expected str, "
+                            f"got {type(name).__name__}: {name!r}")
```
Comment (florimondmanca) [inline httpx/_multipart.py:23]:
```suggestion
            raise TypeError(
                f"Invalid type for name. Expected str, got {type(name)}: {name!r}"
            )
```

File: httpx/_multipart.py [hunk 2]
```diff
@@ -19,9 +19,11 @@ class DataField:
...
         if not isinstance(value, (str, bytes)):
-            raise TypeError("Invalid type for value. Expected str or bytes.")
+            raise TypeError(f"Invalid type for value. Expected str or bytes, "
+                            f"got {type(value).__name__}: {value!r}")
```
Comment (florimondmanca) [inline httpx/_multipart.py:26]:
```suggestion
            raise TypeError(
                f"Invalid type for value. Expected str or bytes, got {type(value)}: {value!r}"
            )
```

**Review by florimondmanca — COMMENTED** [review]
Legit enough, thanks! CI's failing to due formatting, left suggestions that I'll go ahead and commit onto your branch to fix things.

**Review by florimondmanca — APPROVED** [review]

**Comment by plotski** [comment]
Oh, that was what was wrong. I couldn't find an actual error message, just that
the bot was unhappy.

I found some more TypeErrors that didn't include the values that caused them.

Thanks for the quick response.

**Comment by tomchristie** [comment]
Very nice yes. Thank you. 💛

**Pull request merged**